crates/polars-lazy/src/physical_plan/streaming/convert_alp.rs

```diff
@@ -83,21 +81,18 @@ pub(crate) fn insert_streaming_nodes(
     // this allows us to split at joins/unions and share a sink
     let root = insert_file_sink(root, lp_arena);
 
-    // We use mutation to communicate when we need to insert a file sink.
+    // We use a bool flag in the stack to communicate when we need to insert a file sink.
     // This happens for instance when we
     //
     //     ________*non-streamable part of query
     //   /\
     //     ________*streamable below this line so we must insert
     //    /\        a file sink here so the pipeline can be built
     //     /\
-    //
-    // when this is positive we should insert a file sink
-    let mut insert_file_sink_ptr: u32 = 0;
 
     let mut stack = Vec::with_capacity(16);
 
-    stack.push((root, Branch::default(), 0 as CurrentIdx));
+    stack.push((root, Branch::default(), 0 as CurrentIdx, false));
```

Comment:
Now that our stack state is getting more complex. Might be a good time to introduce a `StackFrame` struct.

Reply:
:+1: done